db/db_impl/db_impl_readonly.cc

```diff
@@ -161,6 +174,19 @@ Status DB::OpenForReadOnly(
     const std::vector<ColumnFamilyDescriptor>& column_families,
     std::vector<ColumnFamilyHandle*>* handles, DB** dbptr,
     bool error_if_log_file_exist) {
+  // Attempt to read "CURRENT" file
```

Comment:
Looking good, except we now have code duplication and duplicate I/O checking for file existence when the other overload (no specified column family) is called. Since OpenForReadOnly is a static member function, it should be pretty easy to put the existing functionality of this overload into an anonymous namespace helper function (OpenExistingForReadOnly perhaps), and add another anonymous namespace helper function for this new check.

I'm OK with "extra" `if (!s.ok()) { return s; }`in the callers.

Reply:
Will add an anonymous namespace.
(Previously I thought I may need to modify the function signature and resist doing that :)